packages/patterns/src/components/pattern-schema-quick-navigation.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import {
+	store as blockEditorStore,
+	privateApis as blockEditorPrivateApis,
+} from '@wordpress/block-editor';
+import { useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../lock-unlock';
+import { isOverridableBlock } from '../api';
+
+const { BlockQuickNavigation } = unlock( blockEditorPrivateApis );
+
+export default function PatternSchemaQuickNavigation() {
+	const allClientIds = useSelect(
+		( select ) =>
+			unlock( select( blockEditorStore ) ).getClientIdsWithDescendants(),
+		[]
+	);
```

Comment:
I think `getClientIdsWithDescendants` is a public selector, so unlocking it is unnecessary.

You can also use a filter inside the `mapSelect` if you just return clientId strings. While it's against general policy to avoid `filter/map/etc` iteration inside selectors. I think it's a useful "trick". Here's [a similar example](https://github.com/WordPress/gutenberg/blob/998bd28dc839c8cce7dfead44d8bc276661a7e81/packages/block-editor/src/components/block-inspector/index.js#L41-L45) for the locked block.

_Note to self: it might be helpful to include comments for similar "tricks" explaining why this usage is okay._

Reply:
> I think getClientIdsWithDescendants is a public selector, so unlocking it is unnecessary.

Thanks, not sure why I did that 😄 

> You can also use a filter inside the mapSelect if you just return clientId strings. While it's against general policy to avoid filter/map/etc iteration inside selectors. I think it's a useful "trick". Here's [a similar example](https://github.com/WordPress/gutenberg/blob/998bd28dc839c8cce7dfead44d8bc276661a7e81/packages/block-editor/src/components/block-inspector/index.js#L41-L45) for the locked block.

Interesting. Could you explain more about how that works without causing a performance hit?

In my testing it seems like it would cause a performance problem because `useSelect` runs so regularly even when it's returning the same value.

Reply:
That's true; the `mapSelect` will run whenever the store updates. To be honest, I've not measured performance hits for similar patterns. Though it should be fine for simple filter operations.

I trust to your judgment here :)